TurboWarp/Test/Test.js

```diff
@@ -0,0 +1,619 @@
+          };
+        }
+        return matchCache.arr[args.ITEM - 1] || "";
+      } catch (e) {
+        console.error(e);
+        return "";
+      }
+    }
+
+    countRegex(args, util) {
+      // Fill cache
+      // (ITEM is casted into 0,
+      // but we don't care about the return value)
+      this.matchRegex(args, util);
+      return matchCache.arr.length || 0;
+    }
+
+    testRegex(args, util) {
+      try {
+        args.STRING = args.STRING.toString();
+        args.REGEX = args.REGEX.toString();
+        args.FLAGS = args.FLAGS.toString();
+
+        return new RegExp(args.REGEX, args.FLAGS).test(args.STRING);
+      } catch (e) {
+        console.error(e);
+        return false;
+      }
+    }
+
+    isCase(args, util) {
+      const string = args.STRING.toString();
+      const textCase = args.TEXTCASE.toString();
+      switch (textCase) {
+        case CaseParam.LOWERCASE:
+          return string.toLowerCase() === string;
+        case CaseParam.UPPERCASE:
+          return string.toUpperCase() === string;
+        case CaseParam.MIXEDCASE:
+          return !(
+            string.toUpperCase() === string || string.toLowerCase() === string
+          );
+        case CaseParam.TITLECASE:
+          return string.split(/\b/g).every((word) => {
+            if (!word) return true;
+            const titleCased = word[0].toUpperCase() + word.substring(1);
+            return word === titleCased;
+          });
+        case CaseParam.EXACTTITLECASE:
+          return string.split(/\b/g).every((word) => {
+            if (!word) return true;
+            const titleCased =
+              word[0].toUpperCase() + word.substring(1).toLowerCase();
+            return word === titleCased;
+          });
+        default:
+          return false;
+      }
+    }
+
+    toCase(args, util) {
+      const string = args.STRING.toString();
+      const textCase = args.TEXTCASE.toString();
+      switch (textCase) {
+        case CaseParam.LOWERCASE:
+          return string.toLowerCase();
+        case CaseParam.UPPERCASE:
+          return string.toUpperCase();
+        case CaseParam.MIXEDCASE:
+          return Array.from(string)
+            .map((char, index) =>
+              index % 2 === 0 ? char.toUpperCase() : char.toLowerCase()
+            )
+            .join("");
+        case CaseParam.TITLECASE:
+          return string
+            .split(/\b/g)
+            .map((word) => {
+              if (!word) return "";
+              return word[0].toUpperCase() + word.substring(1);
+            })
+            .join("");
+        case CaseParam.EXACTTITLECASE:
+          return string
+            .split(/\b/g)
+            .map((word) => {
+              if (!word) return "";
+              return word[0].toUpperCase() + word.substring(1).toLowerCase();
+            })
+            .join("");
+        default:
+          return string;
+      }
+    }
+  }
+
+  Scratch.extensions.register(new StringsExt());
```

Comment:
Although Gandi now supports this expanded registration method, this does not configure the cover art and author name, use the tempExt method on the window object